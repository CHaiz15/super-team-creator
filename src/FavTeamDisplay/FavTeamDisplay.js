import React from 'react'
import './FavTeamDisplay.scss'
import { NavLink } from 'react-router-dom'

export const FavTeamDisplay = (props) => {
  let {name, memberOne, memberTwo, memberThree} = props.location.state.team
  return ( 
    <>
      <div className='screen-top'>
        <NavLink type='button' to='/favorites' className='fav-btn'>Favorites</NavLink>
        <h1 className='team-title'>{name}</h1>
      </div>
      <section className='team-container'>
      <section className='members-container'>
        <div  className='member-div'>
          <h2>{memberOne.name}</h2>
          <img className='member-image' src={memberOne.image.url} alt='member one'/>
          <ul className='powers-list'>Stats:
            <li>Intelligence: {memberOne.powerstats.intelligence}</li>
            <li>strength: {memberOne.powerstats.strength}</li>
            <li>Speed: {memberOne.powerstats.speed}</li>
            <li>Durability: {memberOne.powerstats.durability}</li>
            <li>Power: {memberOne.powerstats.power}</li>
            <li>Combat: {memberOne.powerstats.combat}</li>
          </ul>
        </div>
        <div className='member-div'>
          <h2>{memberTwo.name}</h2>
          <img className='member-image' src={memberTwo.image.url} alt='member two'/>
          <ul className='powers-list'>Stats:
            <li>Intelligence: {memberTwo.powerstats.intelligence}</li>
            <li>strength: {memberTwo.powerstats.strength}</li>
            <li>Speed: {memberTwo.powerstats.speed}</li>
            <li>Durability: {memberTwo.powerstats.durability}</li>
            <li>Power: {memberTwo.powerstats.power}</li>
            <li>Combat: {memberTwo.powerstats.combat}</li>
          </ul>
        </div>
        <div className='member-div'>
          <h2>{memberThree.name}</h2>
          <img className='member-image' src={memberThree.image.url} alt='member three'/>
          <ul className='powers-list'>Stats:
            <li>Intelligence: {memberThree.powerstats.intelligence}</li>
            <li>strength: {memberThree.powerstats.strength}</li>
            <li>Speed: {memberThree.powerstats.speed}</li>
            <li>Durability: {memberThree.powerstats.durability}</li>
            <li>Power: {memberThree.powerstats.power}</li>
            <li>Combat: {memberThree.powerstats.combat}</li>
          </ul>
        </div>
      </section>
        <div className='team-stats'>
          <ul>Team Stats:
            <li>Intelligence: {parseInt(memberOne.powerstats.intelligence) + parseInt(memberTwo.powerstats.intelligence) + parseInt(memberThree.powerstats.intelligence)}</li>
            <li>strength: {parseInt(memberOne.powerstats.strength) + parseInt(memberTwo.powerstats.strength) + parseInt(memberThree.powerstats.strength)}</li>
            <li>Speed: {parseInt(memberOne.powerstats.speed) + parseInt(memberTwo.powerstats.speed) + parseInt(memberThree.powerstats.speed)}</li>
            <li>Durability: {parseInt(memberOne.powerstats.durability) + parseInt(memberTwo.powerstats.durability) + parseInt(memberThree.powerstats.durability)}</li>
            <li>Power: {parseInt(memberOne.powerstats.power) + parseInt(memberTwo.powerstats.power) + parseInt(memberThree.powerstats.power)}</li>
            <li>Combat: {parseInt(memberOne.powerstats.combat) + parseInt(memberTwo.powerstats.combat) + parseInt(memberThree.powerstats.combat)}</li>
          </ul>
        </div>
      </section>
    </>
    )
  }

export default FavTeamDisplay