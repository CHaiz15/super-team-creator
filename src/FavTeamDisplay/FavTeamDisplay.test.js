import React from 'react';
import { FavTeamDisplay } from './FavTeamDisplay.js';
import { shallow } from 'enzyme'

describe('FavTeamDisplay', () => {
  let wrapper;
  let expectedProps;

  beforeEach(() => {
    expectedProps = {
      location: {
      state: { 
        team: {
          name:'Best Team Ever',
          memberOne: {
            name:'Hulk', 
            powerstats: {
              Intelligence: 81,
              strength: 40,
              Speed: 29,
              Durability: 55,
              Power: 63,
              Combat: 90,
            },
            image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
          },
          memberTwo: {
            name:'Thanos',
            powerstats: {
              Intelligence: 81,
              strength: 40,
              Speed: 29,
              Durability: 55,
              Power: 63,
              Combat: 90,
            },
            image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
          },
          memberThree: {
            name:'Batman',
            powerstats: {
              Intelligence: 81,
              strength: 40,
              Speed: 29,
              Durability: 55,
              Power: 63,
              Combat: 90,
            },
            image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
          },
          id: 123,
        }
      }
    }
  }
    wrapper = shallow(<FavTeamDisplay {...expectedProps} />);
  })  
  it('should be an instance of the FavTeamDisplay component', () => {
      expect(wrapper).toMatchSnapshot();
  });
})
