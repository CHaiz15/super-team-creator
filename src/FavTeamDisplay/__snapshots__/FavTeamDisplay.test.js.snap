// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavTeamDisplay should be an instance of the FavTeamDisplay component 1`] = `ShallowWrapper {}`;
